packages/babel-plugin/src/utils/css-builders.ts

```diff
@@ -679,9 +679,28 @@ const extractTemplateLiteral = (node: t.TemplateLiteral, meta: Metadata): CSSOut
       isCompiledCSSTaggedTemplateExpression(interpolation, meta.state) ||
       isCompiledCSSCallExpression(interpolation, meta.state);
 
-    if ((!isMidStatement && doesExpressionContainCssBlock) || doesExpressionHaveConditionalCss) {
+    // isTemplateLiteral(nodeExpression) allows handling of expressions
+    // inside template literals inside another template literals
+    // e.g. the output of token function calls by @atlaskit/tokens babel plugin
+    const canBuildExpressionAsCss =
+      (!isMidStatement && doesExpressionContainCssBlock) ||
+      doesExpressionHaveConditionalCss ||
+      t.isTemplateLiteral(nodeExpression);
+
+    if (canBuildExpressionAsCss) {
       // We found something that looks like CSS.
-      const result = buildCss(interpolation, updatedMeta);
+
+      const nestedTemplateLiteralMeta: Metadata = {
+        context: 'fragment',
+        state: updatedMeta.state,
+        parentPath: updatedMeta.parentPath,
+      };
+
+      const buildCssMeta = t.isTemplateLiteral(nodeExpression)
+        ? nestedTemplateLiteralMeta
+        : updatedMeta;
+
+      const result = buildCss(interpolation, buildCssMeta);
```

Comment:
Is it safe to assume that the type of `result.css` is `UnconditionalCssItem` here? (This matters because of line 707 below)

In Jira frontend, `t.isTemplateLiteral(nodeExpression)` and `result.css is not UnconditionalCssItem` are never simultaneously true in this part of the function, but I'm not sure whether it's *theoretically* possible for both to be true

Reply:
To answer my own question: yes, because of `extractObjectExpression`:

https://github.com/atlassian-labs/compiled/blob/0448f52bb0fae6cca0e8217d9662d936eeeb6b1a/packages/babel-plugin/src/utils/css-builders.ts#L530

Code such as

```js
const ListItem = styled.div<{ hide: boolean }>({
    display: (props) => (props.hide ? 'none' : 'inline-block')
});
```

will have the `display` line converted to something like

```js
`display: ${(props) => (props.hide ? 'none' : 'inline-block')}`
```

and the `extractTemplateLiteral` function (the function that this PR modifies) will run afterwards